ReverseLinklist.java

```diff
@@ -0,0 +1,31 @@
+// Time Complexity : algo is iterating through whole list so it will be o(n)
```

Comment:
For recursive solution if you couldn't interpret the solution which maintains the next node and using it to alter the path then try maintaining the previous node in class variables and use it to alter the pointers.

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    
    def __init__(self):
        self.node = None
    def reverseList(self, head: ListNode) -> ListNode:
        if head == None or head.next == None:
            return head
        
        self.helper(head)
        return self.node
    
    def helper(self,head):
        if head.next == None:
            self.node = head
            return head
        
        node = self.helper(head.next)
        
        node.next = head
        head.next = None
        return head
        
Like this